### DISCUSSION TITLE
Limiting the number of resourceTypes to only 2-3 entries/type for a patient for FHIR format

### DISCUSSION THREAD

**Discussion by coldn00dles** [synthetichealth/synthea#1473] [Q&A]
Upon generating data, it can be seen that for each patient the JSON file is too big at times, with one patient having several instances of individual resource types like allergies, claims etc. I want it so that I can limit these to maybe 2-3 instances so the json is relatively smaller. As someone new to synthesia, I just wanted to firstly understand the structure with smaller sizes for a patient and then eventually work with bigger data

**Comment by jawalonoski** [comment]
There is no way to limit the number of specific resources. The goal of the project is to create a coherent medical record, not a "shopping cart" of representative items.

To make smaller files, try limiting the years of history being exported.

You can modify `src/main/resources/synthea.properties` and set the years of history to `1` (the default is 10 years):

```properties
# number of years of history to keep in exported records, anything older than this may be filtered out
# set years_of_history = 0 to skip filtering altogether and keep the entire history
exporter.years_of_history = 1
```

Or on the command line:

```
./run_synthea --exporter.years_of_history=1
```
